libusb/descriptor.c

```diff
@@ -212,7 +216,19 @@ static int parse_interface(libusb_context *ctx,
 		usb_interface->altsetting = altsetting;
 
 		ifp = altsetting + usb_interface->num_altsetting;
+
+		usb_interface->num_altsetting++;
```

Comment:
Moving this above ensures that the altsetting is freed in the case of an error.

Reply:
All comments should be /* */.